### PR TITLE
Fix Syrinx implant

### PR DESCRIPTION
**Pull request by beck-thompson**
## About the PR

Implant was broken, fixed it and made the system a lot cleaner to boot!

## Technical details

Made the transform speaker event get passed through to implants, and also made the implanter use the normal voice mask UI and stuff instead of doing any copy pasting weirdness!

## Media

**Changelog**

:cl:
- fix: Syrinx implant now works correctly

### CODE REVIEW

File: Content.Server/DeltaV/Implants/SyrinxImplantSystem.cs [hunk 1]
```diff
@@ -1,115 +1,30 @@
...
+public sealed class SubdermalBionicSyrinxImplantSystem : EntitySystem
 {
-    [Dependency] private readonly IAdminLogManager _adminLogger = default!;
-    [Dependency] private readonly IPrototypeManager _proto = default!;
-    [Dependency] private readonly SharedPopupSystem _popup = default!;
-    [Dependency] private readonly SharedUserInterfaceSystem _ui = default!;
+    [Dependency] private readonly TagSystem _tag = default!;
+
+    [ValidatePrototypeId<TagPrototype>]
+    private const string BionicSyrinxImplant = "BionicSyrinxImplant";
```
Comment (deltanedas) [inline Content.Server/DeltaV/Implants/SyrinxImplantSystem.cs]: no reason to have this tag check, if you dont want a voice mask implant to work dont make one?

File: Content.Server/DeltaV/Implants/SyrinxImplantSystem.cs [hunk 2]
```diff
@@ -1,115 +1,30 @@
...
-    /// </summary>
-    private void UpdateUI(EntityUid uid, VoiceOverrideComponent voice)
-    {
-        var state = new VoiceMaskBuiState(voice.NameOverride ?? Loc.GetString("voice-mask-default-name-override"), voice.SpeechVerbOverride);
-        _ui.SetUiState(uid, VoiceMaskUIKey.Key, state);
+        SubscribeLocalEvent<VoiceMaskComponent, ImplantImplantedEvent>(OnInsert);
     }
 
-    private void OpenUI(SyrinxImplantSetNameEvent args)
+    private void OnInsert(EntityUid uid, VoiceMaskComponent component, ImplantImplantedEvent args)
```
Comment (deltanedas) [inline Content.Server/DeltaV/Implants/SyrinxImplantSystem.cs]: use `Entity<T>` and `ref` args for all event handlers

File: Content.Shared/Implants/SharedSubdermalImplantSystem.cs [hunk 3]
```diff
@@ -29,6 +30,7 @@ public override void Initialize()
         SubscribeLocalEvent<ImplantedComponent, MobStateChangedEvent>(RelayToImplantEvent);
         SubscribeLocalEvent<ImplantedComponent, AfterInteractUsingEvent>(RelayToImplantEvent);
         SubscribeLocalEvent<ImplantedComponent, SuicideEvent>(RelayToImplantEvent);
+        SubscribeLocalEvent<ImplantedComponent, TransformSpeakerNameEvent>(RelayToImplantEvent);
```
Comment (deltanedas) [inline Content.Shared/Implants/SharedSubdermalImplantSystem.cs]: deltav comment

File: Content.Shared/Implants/SharedSubdermalImplantSystem.cs [hunk 4]
```diff
@@ -1,5 +1,6 @@
 using System.Linq;
 using Content.Shared.Actions;
+using Content.Shared.Chat;
```
Comment (deltanedas) [inline Content.Shared/Implants/SharedSubdermalImplantSystem.cs]: deltav comment

File: Resources/Prototypes/DeltaV/Entities/Objects/Misc/subdermal_implants.yml [hunk 5]
```diff
@@ -9,12 +9,15 @@
...
+      - HarpySinger # Ensure this is only for harpies and if this component gets renamed, CHANGE IT TO THE NEW VALUE!!!
+  - type: VoiceMask
   - type: UserInterface
     interfaces:
       enum.VoiceMaskUIKey.Key:
         type: VoiceMaskBoundUserInterface
-  - type: SyrinxImplant
+  - type: Tag
+    tags:
+    - BionicSyrinxImplant
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Objects/Misc/subdermal_implants.yml]: remove

File: Resources/Prototypes/DeltaV/tags.yml [hunk 6]
```diff
@@ -6,6 +6,9 @@
 - type: Tag
   id: BeltSlotNotBelt #not a 'belt'
 
+- type: Tag
+  id: BionicSyrinxImplant
+
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/tags.yml]: revert all changes to this file including the random whitespace below

File: Content.Server/DeltaV/Implants/SyrinxImplantSystem.cs [hunk 7]
```diff
@@ -1,115 +1,30 @@
...
     {
-        var user = args.Performer;
-        if (!TryComp<VoiceOverrideComponent>(user, out var voice))
+        if (!args.Implanted.HasValue ||
+            !_tag.HasTag(args.Implant, BionicSyrinxImplant))
```
Comment (deltanedas) [inline Content.Server/DeltaV/Implants/SyrinxImplantSystem.cs]: remove tag check

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Content.Server/DeltaV/Implants/SyrinxImplantSystem.cs (hunk 1 above)
Comment (beck-thompson) [reply Content.Server/DeltaV/Implants/SyrinxImplantSystem.cs]: Very good point...

**Comment by beck-thompson** [comment]
Did that fix the weird issue in tags.yml? Commit [3ab54ad](https://github.com/DeltaV-Station/Delta-v/pull/2055/commits/3ab54ad24b080a76ce184610066deaa55c127775) says I added the space but the PR says I didn't... I'm so confused

File: Resources/Prototypes/DeltaV/tags.yml [hunk 8]
```diff
@@ -65,7 +65,7 @@
 
 - type: Tag
   id: HidesHarpyWings
-
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/tags.yml]: the space was *removed* by this pr, realistically not the end of the world but you can always `git checkout usptream/master Resources/Prototypes/DeltaV/tags.yml` for resetting a change you cant see

**Comment by beck-thompson** [comment]
It worked thanks!🫡

**Review by deltanedas — APPROVED** [review]

**Pull request merged**